### DISCUSSION TITLE
regex extractVersion in DockerFile

### DISCUSSION THREAD

**Discussion by plouffej8** [renovatebot/renovate#15981] [Request Help] [answered]
I'm attempting to use extractVersion in a DockerFile with regex to pull out the version number without the v, since it doesnt follow semver. I think the regex should be working to just pull out the version number, but I cant seem to get renovate to notice it as a depedency. It doesnt even show up in the job log. 
```
# renovate: datasource=github-releases depName=taers232c/GAMADV-XTD3 versioning=loose extractVersion=^v(?<version>\d+\.\d+\.\d+)$
ARG GAM_VERSION=6.22.01
```
```
"regexManagers": [
      {
        "description": "Updating annoted package version in Dockerfiles",
        "fileMatch": [
          "(^|/|\\.)Dockerfile$",
          "(^|/)Dockerfile[^/]*$"
        ],
        "matchStrings": [
          "# renovate: datasource=(?<datasource>.+?) depName=(?<depName>.+?) packageName=(?<packageName>.+?)( extractVersion=(?<extractVersion>.+?))?( registryUrl=(?<registryUrl>.+?))?\\s.*_VERSION=\"?(?<currentValue>.+?)\"?\\s"        
        ],
        "extractVersionTemplate": "{{#if extractVersion}}{{{extractVersion}}}{{else}}^v?(?<version>.+)${{/if}}"
      }
    ]
```

Adding or removing versioning=loose doesnt seem to make a difference currently. I feel like I must be missing something obvious, any help?

**Comment by viceice** [comment, marked as answer]
Your regex part for `_VERSION` is probably to greedy, see our own regex manager for sample

https://github.com/renovatebot/.github/blob/b0c3aa85ef2bb242580f20b02b380ca532b4ce17/default.json#L144

**Reply by plouffej8** [reply]
It seems that didnt do that trick, unfortunately, to get it working with the annotation. However! I went and re-implemented my previous attempt at using extractVersion within packageRules and it is now working. Thanks!
